vowpalwabbit/bs.cc

```diff
@@ -233,19 +234,20 @@ void finish_example(vw& all, bs& d, example& ec)
 }
 
 void finish(bs& d)
-{ delete d.pred_vec; }
+{
```

Comment:
My understanding of `BreakBeforeBraces: Allman` for `clang-format` is that every brace has a newline before it. 

For class member functions, there's a separate option that I've set as per @jackgerrits earlier [comment:](https://github.com/VowpalWabbit/vowpal_wabbit/pull/1701#discussion_r245012922) `AllowShortFunctionsOnASingleLine: Inline`.

However, I've realized that the `clang-format-3.4` I installed in CI doesn't have this option, and that the script wasn't running correctly, which I'm close to fixing.

Would there be any objection to updating the Docker image so that we can get a more accessible and up-to-date version of `clang-format`?

Reply:
no objection here.  @jackgerrits ?

Reply:
No objection at all! I recently setup auto-building for our docker image so if you could update the Dockerfile too that would be great

Reply:
So for the short functions, it's okay if they are expanded out over several lines but is it possible for it to be consistent with class functions and free functions?